modules/wpcom-block-editor/class-jetpack-wpcom-block-editor.php

```diff
@@ -58,35 +58,69 @@ public function is_iframed_block_editor() {
 	}
 
 	/**
-	 * Shows a custom message if the user is logged out.
-	 *
-	 * The iframed block editor can be only embedded in WordPress.com if the user is logged
-	 * into the Jetpack site. So we abort the default redirection to the login page (which
-	 * cannot be embedded in a iframe) and instead we explain that we need the user to log
-	 * into Jetpack.
+	 * Prevents frame options header from firing if this is a whitelisted iframe request.
 	 */
-	public function show_error_if_logged_out() {
-		if ( ! get_current_user_id() ) {
-			/* translators: %s: Login URL */
-			$message = __( 'Please <a href="%s" target="_blank" rel="noopener noreferrer">log into</a> your Jetpack-connected site to use the block editor on WordPress.com.', 'jetpack' );
-
-			wp_die(
-				sprintf( wp_kses_post( $message ), esc_url( wp_login_url() ) ),
-				'',
-				array( 'response' => 401 )
-			);
+	public function disable_send_frame_options_header() {
+		if ( $this->framing_allowed() ) {
+			remove_action( 'admin_init', 'send_frame_options_header' );
 		}
 	}
 
 	/**
-	 * Prevents frame options header from firing if this is a whitelisted iframe request.
+	 * Allows to iframe the login page if a user is logged out
+	 * while trying to access the block editor from wordpress.com.
 	 */
-	public function disable_send_frame_options_header() {
-		if ( $this->framing_allowed() ) {
-			remove_action( 'admin_init', 'send_frame_options_header' );
+	public function allow_block_editor_login() {
+		// phpcs:ignore WordPress.Security.NonceVerification
+		if ( empty( $_REQUEST['redirect_to'] ) ) {
+			return;
+		}
+
+		// phpcs:ignore WordPress.Security.NonceVerification
+		$query = wp_parse_url( urldecode( $_REQUEST['redirect_to'] ), PHP_URL_QUERY );
+		$args  = wp_parse_args( $query );
+
+		if ( ! empty( $args['frame-nonce'] ) && $this->framing_allowed( $args['frame-nonce'] ) ) {
+			$_REQUEST['interim-login'] = true;
```

Comment:
Directly manipulating `$_REQUEST` like this feels wrong (like changing `$_GET` query params). Is there any other way to accomplish this?

Reply:
[There is not](https://core.trac.wordpress.org/browser/tags/5.1.1/src/wp-login.php#L486).

Reply:
Well, that's not entirely true. We could add it to the iframe url.